java/server/src/org/openqa/selenium/grid/sessionqueue/RemoveFromSessionQueue.java

```diff
@@ -35,18 +36,20 @@
 
   private final Tracer tracer;
   private final NewSessionQueuer newSessionQueuer;
+  private final RequestId id;
 
-  RemoveFromSessionQueue(Tracer tracer, NewSessionQueuer newSessionQueuer) {
+  RemoveFromSessionQueue(Tracer tracer, NewSessionQueuer newSessionQueuer, RequestId id) {
```

Comment:
Could you please remind me why we use `RemoveFromSessionQueue`? Is there something that prevents us from calling the `remove` action directly in the `NewSessionQueuer`?

Reply:
I thought it might be a good idea to have RemoveFromSessionQueue check if the poll was successful and respond accordingly and to keep that check separate to keep things clean. It also adds a Span for the same to trace the request, in Grid mode the request would be made from the LocalDistributor-> RemoteNewSessionQueuer -> NewSessionQueuerServer.

Reply:
Sounds good to me!